src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.DataGridViewCellAccessibleObject.cs

```diff
@@ -728,9 +732,15 @@ internal override bool IsPatternSupported(UiaCore.UIA patternId)
                 return null;
             }
 
-            internal override int Row => _owner?.OwningRow is not null ? _owner.OwningRow.Index : -1;
+            internal override int Row
+                => _owner?.OwningRow?.Visible is true && _owner.DataGridView is not null
```

Comment:
Consider using a nested property pattern:
```suggestion
                => _owner is { OwningRow.Visible: true, DataGridView: { } }
```

Reply:
I'm not keen on that syntax. @RussKie / @dreddy-work

Reply:
I don't mind this syntax. I don't really like it where it is used as a replacement for null check, but in this situation I think it's ok.

Reply:
I don't think it adds value for the extra cognitive overhead.

Reply:
I agree, it takes time to get used to. This goes for many Rsolyn features we saw over the years. 😄 
I don't have a particularly strong preference in this case (unlike direct null check), so happy either way.